src/boot/uefi.c

```diff
@@ -0,0 +1,272 @@
+#include <runtime.h>
+#include <kernel_machine.h>
+#include <page.h>
+#include <elf64.h>
+#include <pagecache.h>
+#include <storage.h>
+#include <tfs.h>
+#include <uefi.h>
+
+/* Number of spare memory descriptors allocated to cover any additional descriptors that may be
+ * inserted in the memory map between a get_memory_map() call and a exit_boot_services() call. */
+#define UEFI_MEMDESC_SPARE_COUNT    8
+
+//#define UEFI_DEBUG
+#ifdef UEFI_DEBUG
+# define uefi_debug(x, ...) rprintf("UEFI: " x "\n", ##__VA_ARGS__)
+#else
+# define uefi_debug(...)
+#endif
+
+struct efi_guid uefi_block_io_proto = {
+        0x964e5b21, 0x6459, 0x11d2, {0x8e, 0x39, 0x00, 0xa0, 0xc9, 0x69, 0x72, 0x3b}
+};
+struct efi_guid uefi_rng_proto = {
+        0x3152bca5, 0xeade, 0x433d, {0x86, 0x2e, 0xc0, 0x1c, 0xdc, 0x29, 0x1f, 0x44}
```

Comment:
Do these correspond to https://en.wikipedia.org/wiki/GUID_Partition_Table#Partition_type_GUIDs? Would it make sense to use existing Linux' GUID for nanos (as it is already trusted)?

Reply:
No, these are not partition UUIDs, they are UEFI protocol UUIDs, defined in the UEFI spec (see for example https://gitlab.com/ncroxon/gnu-efi/-/blob/master/inc/efiprot.h for these and other protocol-related definitions).